src/librustc_codegen_ssa/mir/constant.rs

```diff
@@ -40,29 +40,37 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         &mut self,
         constant: &mir::Constant<'tcx>,
     ) -> Result<ConstValue<'tcx>, ErrorHandled> {
-        match constant.literal.val {
-            ty::ConstKind::Unevaluated(def_id, substs, promoted) => {
-                let substs = self.monomorphize(&substs);
-                self.cx
-                    .tcx()
-                    .const_eval_resolve(ty::ParamEnv::reveal_all(), def_id, substs, promoted, None)
-                    .map_err(|err| {
-                        if promoted.is_none() {
-                            self.cx
-                                .tcx()
-                                .sess
-                                .span_err(constant.span, "erroneous constant encountered");
-                        }
-                        err
-                    })
-            }
-            ty::ConstKind::Value(value) => Ok(value),
-            _ => {
-                let const_ = self.monomorphize(&constant.literal);
```

Comment:
What you could try is first always `monomorphize`-ing, and *then* checking for `Unevaluated` (you'd need to remove `let substs = self.monomorphize(&substs);`).

It's not the right fix here (see https://github.com/rust-lang/rust/issues/70125#issuecomment-602133708), but it could be useful for handling the erroring constant case.

Reply:
ok, this already seems to fix all tests :thinking: Will still look into your comment on #70125